core/src/main/java/org/elasticsearch/action/search/SearchRequest.java

```diff
@@ -505,18 +361,15 @@ public void writeTo(StreamOutput out) throws IOException {
             out.writeBoolean(true);
             scroll.writeTo(out);
         }
-        out.writeBytesReference(source);
-        out.writeBytesReference(extraSource);
+        if (source == null) {
```

Comment:
I got confused with the count request, I see that this can be null, but I still wonder if it should? When does it make sense for it to be null? If it wasn't we could drop a lot of null checks around that.